src/services/api/FirebaseApi.ts

```diff
@@ -551,8 +551,10 @@ export class FirebaseApi implements ServiceApi {
     };
     console.log("playedResult", result.lesson.id, JSON.stringify(playedResult));
 
-    this._studentResultCache[student.docId].lessons[result.lesson.id] =
-      playedResult;
+    if (this._studentResultCache[student.docId] !== undefined) {
```

Comment:
here you skipped the statement by if condition... you shouldn't do like this... you have to initialize **this._studentResultCache[student.docId]**